File: HACKERRANK/Problem Solving/Algorithms/Implementation/Jumping on the Clouds/jumpingOnClouds.py
from typing import List


def jumpingOnClouds(n: int, cloud_arr: List[int]) -> int:
    current_pos, num_jumps = 0, 0
    
    while current_pos < n:

        if current_pos == n - 1:
            break;

        if current_pos == n - 2:
            if cloud_arr[current_pos + 1] == 0:
                num_jumps += 1
                break;
            else:
                break;

        if cloud_arr[current_pos + 2] == 0:
            current_pos += 2
            num_jumps += 1
        else:
            current_pos += 1
            num_jumps += 1
    
    return num_jumps

if __name__ == "__main__":
    print(jumpingOnClouds(7, [0, 1, 0, 0, 0, 1, 0]))
    print(jumpingOnClouds(6, [0, 0, 0, 1, 0, 0]))

File: HACKERRANK/Problem Solving/Algorithms/Implementation/Jumping on the Clouds/jumpingOnClouds.c
#include<stdio.h>

int jumpingOnClouds(int len, int arr[]) {
    int pos = 0;
    int num_jumps = 0;

    while (pos < len) {
        if (pos == len - 1) {
            break;
        }

        if (pos == len - 2) {
            if (arr[pos + 1] == 0) {
                num_jumps++;
                break;
            } else {
                break;
            }
        }

        if (arr[pos + 2] == 0) {
            pos += 2;
            num_jumps++;
        } else {
            pos += 1;
            num_jumps++;
        }

    }

    return num_jumps;
}

int main() {
    int arr1[] = {0, 1, 0, 0, 0, 1, 0};
    printf("%d\n", jumpingOnClouds(7, arr1));

    int arr2[] = {0, 0, 0, 1, 0, 0};
    printf("%d\n", jumpingOnClouds(6, arr2));

}